File: airbyte-connector-builder-server/src/main/java/io/airbyte/connector_builder/file_writer/AirbyteArgument.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connector_builder.file_writer;

import java.io.IOException;

/**
 * Exposes a way of constructing arguments for Airbyte protocol commands.
 */
public class AirbyteArgument {

  private String filepath;
  private final AirbyteFileWriter writer;

  public AirbyteArgument(final AirbyteFileWriter writer) {
    this.writer = writer;
  }

  public void setUpArg(final String name, final String contents) throws IOException {
    final String filepath = this.writer.write(name, contents);
    this.setFilepath(filepath);
  }

  public String getFilepath() {
    return this.filepath;
  }

  public void setFilepath(final String filepath) {
    this.filepath = filepath;
  }

}

File: airbyte-connector-builder-server/src/main/kotlin/io/airbyte/connectorbuilder/filewriter/AirbyteArgument.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connectorbuilder.filewriter

import java.io.IOException

/**
 * Exposes a way of constructing arguments for Airbyte protocol commands.
 */
class AirbyteArgument(
  private val writer: AirbyteFileWriter,
) {
  var filepath: String? = null

  @Throws(IOException::class)
  fun setUpArg(
    name: String,
    contents: String,
  ) {
    val filepath = writer.write(name, contents)
    this.filepath = filepath
  }
}
